core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java

```diff
@@ -235,15 +247,29 @@ public static void populate() {
 
                 BlockDefinition bedrockBlock = null;
                 Integer firstBlockRuntimeId = entry.getValue().getFirstBlockRuntimeId();
+                BlockDefinition customBlockItemOverride = null;
                 if (firstBlockRuntimeId != null) {
                     BlockDefinition blockOverride = bedrockBlockIdOverrides.get(bedrockIdentifier);
-                    if (blockOverride != null) {
+
+                    // We'll do this here for custom blocks we want in the creative inventory so we can piggyback off the existing logic to find these
+                    // blocks in creativeItems
+                    CustomBlockData customBlockData = BlockRegistries.CUSTOM_BLOCK_ITEM_OVERRIDES.getOrDefault(javaItem.javaIdentifier(), null);
+                    if (customBlockData != null) {
+                        // this block has a custom item override and thus we should use its runtime ID for the ItemMapping
+                        if (customBlockData.includedInCreativeInventory()) {
+                            CustomBlockState customBlockState = customBlockData.defaultBlockState();
+                            customBlockItemOverride = blockMappings.getCustomBlockStateDefinitions().getOrDefault(customBlockState, null);
+                        }
+                    }
+
+                    // If it' s a custom block we can't do this because we need to make sure we find the creative item
+                    if (blockOverride != null && customBlockItemOverride == null) {
                         // Straight from BDS is our best chance of getting an item that doesn't run into issues
                         bedrockBlock = blockOverride;
                     } else {
                         // Try to get an example block runtime ID from the creative contents packet, for Bedrock identifier obtaining
                         int aValidBedrockBlockId = blacklistedIdentifiers.getOrDefault(bedrockIdentifier, -1);
-                        if (aValidBedrockBlockId == -1) {
+                        if (aValidBedrockBlockId == -1 && customBlockItemOverride == null) {
```

Comment:
This `customBlockItemOverride == null` check means that `aValidBedrockBlockId` can be `-1` in the `else` below making `blockMappings.getDefinition(aValidBedrockBlockId)` in line `282` return `null` and throw an exception:

```
[12:12:20 ERROR]: Could not pass event ServerLoadEvent to Geyser-Spigot v2.1.2-SNAPSHOT
java.lang.ExceptionInInitializerError: null
	at org.geysermc.geyser.GeyserImpl.initialize(GeyserImpl.java:192) ~[Geyser-Spigot.jar:?]
	at org.geysermc.geyser.GeyserImpl.start(GeyserImpl.java:710) ~[Geyser-Spigot.jar:?]
	at org.geysermc.geyser.platform.spigot.GeyserSpigotPlugin.postStartup(GeyserSpigotPlugin.java:216) ~[Geyser-Spigot.jar:?]
	at org.geysermc.geyser.platform.spigot.GeyserSpigotPlugin$1.onServerLoaded(GeyserSpigotPlugin.java:185) ~[Geyser-Spigot.jar:?]
	at com.destroystokyo.paper.event.executor.MethodHandleEventExecutor.execute(MethodHandleEventExecutor.java:40) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:git-Paper-448]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[paper-1.19.3.jar:git-Paper-448]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[paper-1.19.3.jar:git-Paper-448]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:615) ~[paper-api-1.19.3-R0.1-SNAPSHOT.jar:?]
	at net.minecraft.server.MinecraftServer.loadWorld0(MinecraftServer.java:636) ~[paper-1.19.3.jar:git-Paper-448]
	at net.minecraft.server.MinecraftServer.loadLevel(MinecraftServer.java:434) ~[paper-1.19.3.jar:git-Paper-448]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:308) ~[paper-1.19.3.jar:git-Paper-448]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1101) ~[paper-1.19.3.jar:git-Paper-448]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:316) ~[paper-1.19.3.jar:git-Paper-448]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "org.geysermc.geyser.registry.type.GeyserBedrockBlock.getState()" because the return value of "org.geysermc.geyser.registry.type.BlockMappings.getDefinition(int)" is null
	at org.geysermc.geyser.registry.populator.ItemRegistryPopulator.populate(ItemRegistryPopulator.java:282) ~[Geyser-Spigot.jar:?]
	at org.geysermc.geyser.registry.Registries.<clinit>(Registries.java:176) ~[Geyser-Spigot.jar:?]
	... 16 more
```

This most likely needs some form of of additional querying of the identifier from the customBlockItemOverride or something.

Reply:
What block were you trying to override here that triggered this condition?

Reply:
Added a log for `javaItem.javaIdentifier()` when `blockMappings.getDefinition(aValidBedrockBlockId) == null`:

minecraft:carved_pumpkin
minecraft:jack_o_lantern
minecraft:white_glazed_terracotta
minecraft:orange_glazed_terracotta
minecraft:magenta_glazed_terracotta
minecraft:light_blue_glazed_terracotta
minecraft:yellow_glazed_terracotta
minecraft:lime_glazed_terracotta
minecraft:pink_glazed_terracotta
minecraft:gray_glazed_terracotta
minecraft:light_gray_glazed_terracotta
minecraft:cyan_glazed_terracotta
minecraft:purple_glazed_terracotta
minecraft:blue_glazed_terracotta
minecraft:brown_glazed_terracotta
minecraft:green_glazed_terracotta
minecraft:red_glazed_terracotta
minecraft:black_glazed_terracotta
minecraft:redstone_block
minecraft:pumpkin
minecraft:carved_pumpkin
minecraft:jack_o_lantern
minecraft:magenta_carpet
minecraft:light_blue_carpet
minecraft:yellow_carpet
minecraft:pink_carpet
minecraft:gray_carpet
minecraft:light_gray_carpet
minecraft:blue_carpet
minecraft:red_carpet
minecraft:white_glazed_terracotta
minecraft:orange_glazed_terracotta
minecraft:magenta_glazed_terracotta
minecraft:light_blue_glazed_terracotta
minecraft:yellow_glazed_terracotta
minecraft:lime_glazed_terracotta
minecraft:pink_glazed_terracotta
minecraft:gray_glazed_terracotta
minecraft:light_gray_glazed_terracotta
minecraft:cyan_glazed_terracotta
minecraft:purple_glazed_terracotta
minecraft:blue_glazed_terracotta
minecraft:brown_glazed_terracotta
minecraft:green_glazed_terracotta
minecraft:red_glazed_terracotta
minecraft:black_glazed_terracotta
minecraft:redstone_block
minecraft:pumpkin
minecraft:carved_pumpkin
minecraft:jack_o_lantern
minecraft:magenta_carpet
minecraft:light_blue_carpet
minecraft:yellow_carpet
minecraft:pink_carpet
minecraft:gray_carpet
minecraft:light_gray_carpet
minecraft:blue_carpet
minecraft:red_carpet
minecraft:white_glazed_terracotta
minecraft:orange_glazed_terracotta
minecraft:magenta_glazed_terracotta
minecraft:light_blue_glazed_terracotta
minecraft:yellow_glazed_terracotta
minecraft:lime_glazed_terracotta
minecraft:pink_glazed_terracotta
minecraft:gray_glazed_terracotta
minecraft:light_gray_glazed_terracotta
minecraft:cyan_glazed_terracotta
minecraft:purple_glazed_terracotta
minecraft:blue_glazed_terracotta
minecraft:brown_glazed_terracotta
minecraft:green_glazed_terracotta
minecraft:red_glazed_terracotta
minecraft:black_glazed_terracotta
minecraft:white_concrete
minecraft:orange_concrete
minecraft:magenta_concrete
minecraft:light_blue_concrete
minecraft:yellow_concrete
minecraft:lime_concrete
minecraft:pink_concrete
minecraft:gray_concrete
minecraft:light_gray_concrete
minecraft:cyan_concrete
minecraft:purple_concrete
minecraft:blue_concrete
minecraft:brown_concrete
minecraft:green_concrete
minecraft:red_concrete
minecraft:black_concrete
minecraft:redstone_block